### DISCUSSION TITLE
Github action gone horribly wrong

### DISCUSSION THREAD

**Discussion by andreyvolokitin** [community/community#25745] [Actions]
<p>My action:</p>
<ul>
<li>has a name equal to the action file path in the github, instead of the actual <code>name</code> attribute</li>
<li>triggers on every <code>push</code> event, despite the fact that there was no <code>push</code> event defined in the action</li>
<li>fails to run while doing above, despite that locally it works while testing with <code>nektos/act</code>
</li>
<li>does not run on the actual event which I define in the action (release/published)</li>
</ul>
<p>The action seems like formatted properly, and it works locally. I have absolutely no clue what is going on… Do you have any suggestions?</p>
<p>Here’s my action: <a href="https://gist.github.com/andreyvolokitin/5eaf42380a17f7eed2c992573a3859ad" class="inline-onebox" rel="noopener nofollow ugc">release.yml · GitHub</a><br>
Here’s how it looks in gihub actions panel (it also sends me an email about the fail with a description of “No jobs were run”): <a href="https://i.imgur.com/WngXcaY.png" rel="noopener nofollow ugc">https://i.imgur.com/WngXcaY.png</a></p>

**Comment by AF250329** [comment]
<p>It just a guess - but may be you editing the wrong file ? For example when I create a new file in <code>.github</code> folder actual file created in root solution folder - because <code>.github</code> it is not an actual folder for VS… So you think you editing one file but actually it is not in correct (very important) folder ? Because the gist look correct - it should not run on ‘push’.</p>
<p>About ‘fails to run’ - please post error message or some output - because it hard to guess what do you mean by ‘fails’…</p>

**Comment by ChristopherHX** [comment, marked as answer]
<p>You have a syntax error in your workflow, GitHub informs you if you push invalid workflow files.</p>
<blockquote>
<p>The template is not valid. release.yml (Line: 66, Col: 9): There’s not enough info to determine what you meant. Add one of these properties: run, shell, uses, with, working-directory</p>
</blockquote>
<p>GitHub Actions and my nektos/act replacement ( <a href="https://github.com/ChristopherHX/runner.server" class="inline-onebox" rel="noopener nofollow ugc">GitHub - ChristopherHX/runner.server: An unofficial Console Client and Server for GitHub Action Runners</a> ) show this error.</p>
<p>Replace the hyphen with a space ( Line 68 ) <a href="https://gist.github.com/andreyvolokitin/5eaf42380a17f7eed2c992573a3859ad#file-release-yml-L68" class="inline-onebox" rel="noopener nofollow ugc">release.yml · GitHub</a></p>
<p>Invalid</p>
<pre><code class="lang-yaml">      - name: Get dist
      - uses: actions/download-artifact@master
        with:
          name: dist
          path: dist
</code></pre>
<p>Valid</p>
<pre><code class="lang-yaml">      - name: Get dist
        uses: actions/download-artifact@master
        with:
          name: dist
          path: dist
</code></pre>
<p>You might want to check your workflow files with <a href="https://github.com/rhysd/actionlint" class="inline-onebox" rel="noopener nofollow ugc">GitHub - rhysd/actionlint: Static checker for GitHub Actions workflow files</a> before pushing them.</p>

**Comment by andreyvolokitin** [comment]
<p>Thank you! I knew it was something obvious, linting is a truly must</p>
<p>Edit: what I saw is that there was no notes about malformed action, I pushed it via PR, github just picked it up as an empty file, so it didn’t get the name, used the file path instead, and I guess all actions run on push by default — that explains it all. But there was no warning about the fact that github wasn’t able to recognize the contents of the file properly</p>
